### PR TITLE
1.8 release

### PR DESCRIPTION
**Pull request by Diman0**
## What type of PR?

1.8 release.

## What does this PR do?
Final changes required for the 1.8 release.

### Related issue(s)
- #1829 can be closed after this PR is backported.

## Prerequistes
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by Diman0** [comment]
There was a merge conflict that I forgot to resolve. I resolved it. If bors fails I expect it works when we retry the bors try.

**Comment by Diman0** [comment]
bors try

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,18 +4,40 @@ Changelog
...
+
+- Features: Update version of roundcube webmail and carddav plugin. This is a security update. ([#1841](https://github.com/Mailu/Mailu/issues/1841))
+- Features: Update version of rainloop webmail to 1.16.0. This is a security update. ([#1845](https://github.com/Mailu/Mailu/issues/1845))
+- Features: Changed default value of AUTH_RATELIMIT_SUBNET to false. Increased default value of the rate limit in setup utility (AUTH_RATELIMIT) to a higher value. ([#1867](https://github.com/Mailu/Mailu/issues/1867))
+- Features: Update jquery used in setup. Set pinned versions in requirements.txt for setup. This is a security update. ([#1880](https://github.com/Mailu/Mailu/issues/1880))
+- Bugfixes: Replace PUBLIC_HOSTNAME and PUBLIC_IP in "Received" headers to ensure that no undue spam points are attributed ([#191](https://github.com/Mailu/Mailu/issues/191))
+- Bugfixes: Don't replace nested headers (typically in attached emails) ([#1660](https://github.com/Mailu/Mailu/issues/1660))
+- Bugfixes: Fix letsencrypt access to certbot for the mail-letsencrypt flavour ([#1686](https://github.com/Mailu/Mailu/issues/1686))
+- Bugfixes: Fix CVE-2020-25275 and CVE-2020-24386 by using alpine 3.13 for
```
Comment (nextgens) [inline CHANGELOG.md:30]: s/using alpine 3.13/upgrading alpine/

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,18 +4,40 @@ Changelog
...
+  dovecot which contains a fixed dovecot version. ([#1720](https://github.com/Mailu/Mailu/issues/1720))
+- Bugfixes: Antispam service now uses a static hostname. Rspamd history is only retained when the service has a fixed hostname. ([#1837](https://github.com/Mailu/Mailu/issues/1837))
+- Bugfixes: Fix a bug preventing colons from being used in passwords when using radicale/webdav. ([#1861](https://github.com/Mailu/Mailu/issues/1861))
+- Bugfixes: Remove dot in blueprint name to prevent critical flask startup error in setup. ([#1874](https://github.com/Mailu/Mailu/issues/1874))
+- Bugfixes: fix punycode encoding of domain names ([#1891](https://github.com/Mailu/Mailu/issues/1891))
+- Improved Documentation: Update fail2ban documentation to use systemd backend instead of filepath for journald ([#1857](https://github.com/Mailu/Mailu/issues/1857))
+- Misc:  ([#1783](https://github.com/Mailu/Mailu/issues/1783))
```
Comment (nextgens) [inline CHANGELOG.md:37]: Do we want to go into details here? We should encourage people to rekey their SECRET_KEY... but may ask them to do it again in 1.9

File: docs/releases.rst [hunk 3]
```diff
@@ -1,8 +1,69 @@
...
+- Japanese localisation is now available.
+
+
+Upgrading
+`````````
+
+Upgrade should run fine as long as you generate a new compose or stack
+configuration and upgrade your mailu.env.
+
+Please not that the shipped image for PostgreSQL database is deprecated.
```
Comment (nextgens) [inline docs/releases.rst:44]: s/ not / note /

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,18 +4,40 @@ Changelog
...
 If there are local override files, they will need to be moved from overrides/ to overrides/dovecot and overrides/postfix/.
 See https://mailu.io/1.8/faq.html#how-can-i-override-settings for all the mappings.
 
-Please note that the shipped image for PostgreSQL database is deprecated.
-We advise to switch to an external database server.
+One major change for the docker compose file is that the antispam container needs a fixed hostname [#1837](https://github.com/Mailu/Mailu/issues/1837).
+This is handled when you regenerate the docker-compose file. A fixed hostname is required to retain rspamd history. 
+This is also handled in the helm-chart repo.
+
+Please not that the shipped image for PostgreSQL database is deprecated.
```
Comment (nextgens) [inline CHANGELOG.md:16]: s/ not / note /

File: docs/releases.rst [hunk 5]
```diff
@@ -1,8 +1,69 @@
...
+
+New Functionality & Improvements
+````````````````````````````````
+
+Here’s a short summary of new features:
+
+- Roundcube and Rainloop have been updated.
+- All dependencies have been updated to the latest security update.
+- Fail2ban documentation has been improved.
+- Switch from client side (cookie) sessions to server side sessions.
```
Comment (nextgens) [inline docs/releases.rst:24]: and protect against session-fixation attacks. We recommend that you change your SECRET_KEY after upgrading.

File: CHANGELOG.md (hunk 2 above)
Comment (Diman0) [reply CHANGELOG.md:37]: To what item are you referring? The misc item #1783? I don't see in the PR that it is advised to recreate the SECRET_KEY.
But creating a new uppercase alphanumeric string of length 16 should be sufficient?
After that I reckon the containers must be recreated. I could add to the CHANGELOG,

Security improvements have been made to Mailu. To receive the full benefits of these improvements, we recommend to recreate your SECRET_KEY in mailu.env. This happens when you recreate your docker-compose.yml & mailu.env file via the [setup utility](setup.mailu.io). You can manually create a new secret key by using the following command on linux:
```cat /dev/urandom | tr -dc 'A-Z0-9' | fold -w ${1:-16} | head -n 1```
All containers must be recreated after changing the SECRET_KEY.

Comment (nextgens) [reply CHANGELOG.md:37]: Yes that's good ... it's just about making it explicit for those that use k8s

**Comment by Diman0** [comment]
I made all recommended changes. I increased the waiting timeout for starting all containers in each test by 1 minute. I hope this is sufficient. 

bors try

**Review by nextgens — APPROVED** [review]
LGTM

**Comment by Diman0** [comment]
@Mergifyio backport

**Comment by nextgens** [comment]
@Mergifyio backport 1.8

**Pull request merged**